src/bootstrap/test.rs

```diff
@@ -1105,6 +1105,31 @@ impl Step for Compiletest {
             }
         }
 
+        let clang_version = Command::new(&clang_exe)
+            .arg("--version")
+            .output()
+            .map(|output| { String::from_utf8_lossy(&output.stdout).to_string() })
+            .ok();
+        if let Some(ref vers) = clang_version {
+            cmd.arg("--clang-version").arg(vers);
+        }
+
+        if let Some(var) = env::var_os("RUSTBUILD_FORCE_CLANG_BASED_TESTS") {
+            match &var.to_string_lossy()[..] {
+                "1" | "yes" | "on" => {
```

Comment:
Heh since this is an internal env var I think matching on only "yes" or "1" is probably fine, we tend to not really need that much error handling and support for internal features like this

Reply:
I wanted to try and make it less likely to have a typo causing things to be silently ignored. What would be really needed though would be fuzzy matching on env var *names*...

Reply:
True yeah, although we could also just check for existence here instead of checking for value perhaps?

Reply:
That's what wanted to do at first but then I remembered the `CARGO_INCREMENTAL=0` confusion. Since it's not really complicated to support a few different options here, I just did that.